### ISSUE TITLE
Wrapper Exceptions accumulate modifications.

### ISSUE THREAD

**Issue by DanFTRX** [MindscapeHQ/raygun4net#373]
In Raygun4net4 when a wrapper exception is split up into its respective exceptions, it passes the same collection instances to each of the new messages:
https://github.com/MindscapeHQ/raygun4net/blob/master/Mindscape.Raygun4Net4/RaygunClient.cs#L440

The documentation for modifying messages suggests that we are able to modify these messages during the SendMessage event:
https://raygun.com/docs/languages/net/raygun.com#modify-cancel-message

But if we have a modifier, such as a tag that we append, the tags list gets modified for each and every exception in the AggregateException, but accumulating because they all share the same instance.

To summarize, the RaygunMessageBuilder needs to make copies of the lists/collections added to it otherwise AggregateExceptions and other exceptions that wrap multiple exceptions will accumulate modifications unnecessarily.

**Comment by QuantumNightmare** [comment]
Hi DanFTRX,

Thanks for pointing this out. We'll take a look at this for a future release of Raygun4Net.

-Jason Fauchelle

**Comment by DanFTRX** [comment]
Any word on this?
